src/Silex/Application.php

```diff
@@ -605,8 +605,14 @@ public function run(?Request $request = null)
      *
      * If you call this method directly instead of run(), you must call the
      * terminate() method yourself if you want the finish filters to be run.
+     *
+     * @param \Symfony\Component\HttpFoundation\Request $request
+     * @param int $type
+     * @param bool $catch
+     *
+     * @return \Symfony\Component\HttpFoundation\Response
      */
-    public function handle(Request $request, $type = HttpKernelInterface::MASTER_REQUEST, $catch = true)
+    public function handle(Request $request, int $type = HttpKernelInterface::MASTER_REQUEST, bool $catch = true): Response
```

Comment:
It won't work for `symfony/http-kernel` < 5
See https://github.com/symfony/http-kernel/blob/v4.4.37/HttpKernelInterface.php

The same issue with routing-related changes. Please verify.